### PR TITLE
de: continued German translation until chapter 6.5.3

### PR DESCRIPTION
**Pull request by Throvn**

### CODE REVIEW

File: po/de.po [hunk 1]
```diff
@@ -1888,10 +1888,10 @@ msgid ""
 "writing, the latest stable Rust release has these version numbers:"
 msgstr ""
 "Wenn Sie mit dem Code auf Ihrem eigenen System experimentieren möchten, sollten Sie\n"
-"zuerst Rust zu installieren. Befolgen Sie dazu die [Anweisungen in dem Rust\n"
+"zuerst Rust installieren. Befolgen Sie dazu die [Anweisungen in dem Rust\n"
```
Comment (ronaldfw) [inline po/de.po:1891]: We decided at some point that we want to use the informal "Du" in the training and not "Sie". Please change here and in the rest of the translation, thanks!

File: po/de.po [hunk 2]
```diff
@@ -2997,33 +3033,38 @@ msgid ""
...
 msgstr ""
-"* Das `format!()`-Makro ist eine bequeme Möglichkeit, einen eigenen String aus dynamischen Werten "
-"zu generieren. Es\n"
-"  akzeptiert die gleiche Formatspezifikation wie `println!()`.\n"
+"* `&str` führt einen String-Slice ein, der eine unveränderliche Referenz auf UTF-8-codierte String-"
+"Daten ist, die in einem Speicherblock gespeichert sind. String-Literale (`\"Hallo\"`) werden in "
+"der Binärdatei des Programms gespeichert.\n"
+"\n"
+"* Der `String`-Typ von Rust ist eine Verpackung (wrapper) um einen Byte-Vektor. Wie bei einem "
+"`Vec<T>` ist es im Besitz.\n"
```
Comment (ronaldfw) [inline po/de.po:3041]: The sentence "Wie bei einem Vec ist es im Besitz." sounds strange to me, it should state what it owns.

File: po/de.po [hunk 3]
```diff
@@ -3101,6 +3151,10 @@ msgid ""
 "[rustdoc](https://doc.rust-lang.org/rustdoc/what-is-rustdoc.html) tool. It is\n"
 "idiomatic to document all public items in an API using this pattern."
 msgstr ""
+"Die Inhalte werden als Markdown behandelt. Alle veröffentlichten Rust-Bibliothekskästen (library "
+"cates) werden automatisch unter [`docs.rs`](https://docs.rs) mit [rustdoc](https://doc.rust-lang."
```
Comment (ronaldfw) [inline po/de.po:3155]: Typo: cates -> crates

**Review by ronaldfw — APPROVED** [review]
Thanks. Overall looks good to me, please fix the smaller issues before merging.

**Comment by Throvn** [comment]
I will apply the smaller changes tonight, so you should be able to merge tomorrow.
Is it okey if I replace all "Sie" with "Du" in this PR (also the ones from previouse PRs)?

**Comment by ronaldfw** [comment]
> I will apply the smaller changes tonight, so you should be able to merge tomorrow. Is it okey if I replace all "Sie" with "Du" in this PR (also the ones from previouse PRs)?

Yes, would be great if you'd replace "Sie" with "Du" in this PR here.

File: po/de.po (hunk 2 above)
Comment (Throvn) [reply po/de.po:3040]: Tbh, I also need some clarification here (from the english language). Is it ment that the scope owns the `Vec` or that the `Vec` owns the entries?

Comment (mgeisler) [reply po/de.po:3040]: > Tbh, I also need some clarification here (from the english language). Is it ment that the scope owns the `Vec` or that the `Vec` owns the entries?

Sorry about that, we should clean up the English text to make it clearer!

Here, the difference that the text tries to highlight is that a `&str` value is _borrowed_. When you have `let x: &str = "foo bar"`, the `x` variable does not own the string data. As a consequence, the string data is not deallocated when `x` goes out of scope.

In contrast, a `String` value is _owned_. When you do `let x: String = String::from("foo bar")`, the `x` variable owns the `String`, which internally owns a `Vec<u8>`, which owns every `u8` inside it. So transitively, `x` owns the string data and when `x` goes out of scope, the string data is deallocated.

We can probably make https://github.com/google/comprehensive-rust/blob/main/src/basic-syntax/string-slices.md clearer about this :smile:

File: po/de.po [hunk 4]
```diff
@@ -1769,9 +1769,9 @@ msgid ""
 "      * It is also extensible with sub command plugins as well (such as [cargo clippy]).\n"
 "  * Read more from the [official Cargo Book]"
 msgstr ""
-"* Rust hat einen schnellen Veröffentlichungsplan. Alle sich Wochen wird eine neue Version "
-"veröffentlicht. Neue Versionen bleiben rückwärtskompatibel mit\n"
-"   Älteren Versionen und sie fügen neue Funktionen hinzu.\n"
+"* Rust hat einen schnellen Veröffentlichungsplan. Alle sechs Wochen wird eine neue Version "
+"veröffentlicht. Neue Versionen bleiben rückwärtskompatibel mit Älteren Versionen und fügen neue "
```
Comment (ronaldfw) [inline po/de.po:1773]:
```suggestion
"veröffentlicht. Neue Versionen bleiben rückwärtskompatibel mit älteren Versionen und fügen neue "
```

File: po/de.po [hunk 5]
```diff
@@ -1514,8 +1514,8 @@ msgid ""
 "go. You can then copy/paste the examples into `src/main.rs` to experiment with\n"
 "them:"
 msgstr ""
-"Sie sollten eine neue Kiste (Crate) einrichten und die Abhängigkeiten (Dependencies) herunterladen "
-"und einsatzbereit machen. Anschließend können Sie die Beispiele kopieren und in „src/main.rs“ "
+"Du solltest eine neue Kiste (Crate) einrichten und die Abhängigkeiten (Dependencies) herunterladen "
```
Comment (ronaldfw) [inline po/de.po:1517]:
```suggestion
"Du solltest eine neue Kiste (crate) einrichten und die Abhängigkeiten (dependencies) herunterladen "
```

File: po/de.po [hunk 6]
```diff
@@ -1783,14 +1783,14 @@ msgstr ""
 "\n"
 "   * Editionen dürfen rückwärtsinkompatible Änderungen vornehmen.\n"
 "\n"
-"   * Um zu verhindern, dass Code nicht kaputt geht, sind Editionen optional: Sie wählen ihre "
-"Edition für ihre Kiste (Crate) in der Datei \"Cargo.toml\"\n"
+"   * Um zu verhindern, dass Code nicht kaputt geht, sind Editionen optional: Du wählst ihre "
+"Edition für deine Kiste (Crate) in der Datei `Cargo.toml`\n"
```
Comment (ronaldfw) [inline po/de.po:1787]:
```suggestion
"Edition für deine Kiste (crate) in der Datei `Cargo.toml`\n"
```

File: po/de.po [hunk 7]
```diff
@@ -1822,10 +1822,10 @@ msgid ""
 "exercises. On the last day, we will do a larger exercise which shows you how to\n"
 "work with dependencies and for that you need Cargo."
 msgstr ""
-"Die Installation von Cargo wird dennoch empfohlen: Es wird Ihnen die Bearbeitung der Übungen "
+"Die Installation von Cargo wird dennoch empfohlen: Es wird Dir die Bearbeitung der Übungen "
 "erleichtern.\n"
-"Am letzten Tag werden wir eine größere Übung machen, die Ihnen zeigt, wie\n"
-"Sie mit Abhängigkeiten (Dependencies) arbeiten und dafür benötigen Sie Cargo."
+"Am letzten Tag werden wir eine größere Übung machen, die zeigt, wie\n"
+"Du mit Abhängigkeiten (Dependencies) arbeitest und dafür benötigst Du Cargo."
```
Comment (ronaldfw) [inline po/de.po:1828]:
```suggestion
"Du mit Abhängigkeiten (dependencies) arbeitest und dafür benötigst Du Cargo."
```

File: po/de.po [hunk 8]
```diff
@@ -2004,7 +2044,7 @@ msgstr ""
 
 #: src/welcome-day-1.md:16
 msgid "Please remind the students that:"
-msgstr "Bitte erinnern Sie die Schüler daran:"
+msgstr "Bitte erinnere die Schüler daran:"
```
Comment (ronaldfw) [inline po/de.po:2047]:
```suggestion
msgstr "Bitte erinnere die Kursteilnehmer daran:"
```

File: po/de.po [hunk 9]
```diff
@@ -2040,14 +2080,14 @@ msgstr ""
 "Die Idee für den ersten Tag ist, Rust _gerade genug_ zu zeigen, um über den berühmten "
 "Ausleihenprüfer (borrow checker) sprechen zu können.\n"
 "Die Art und Weise, wie Rust mit Speicher umgeht, ist ein wichtiges Merkmal\n"
-"und das sollten Sie den Schülern direkt zeigen."
+"und das solltest Du den Schülern direkt zeigen."
```
Comment (ronaldfw) [inline po/de.po:2083]:
```suggestion
"und das solltest Du den Kursteilnehmern direkt zeigen."
```

File: po/de.po [hunk 10]
```diff
@@ -2233,8 +2273,7 @@ msgid ""
 "inputs."
 msgstr ""
 "Der Code implementiert die Collatz-Vermutung: Es wird angenommen, dass die Schleife immer enden "
-"wird, allerdings ist das noch nicht bewiesen. Ändern Sie den Code und spielen Sie mit "
-"verschiedenen Eingaben."
+"wird, allerdings ist das noch nicht bewiesen. Ändere den Code und spiel mit verschiedenen Eingaben."
```
Comment (ronaldfw) [inline po/de.po:2276]:
```suggestion
"wird, allerdings ist das noch nicht bewiesen. Ändere den Code und spiele mit verschiedenen Eingaben."
```

File: po/de.po [hunk 11]
```diff
@@ -2253,21 +2292,21 @@ msgid ""
...
 "\n"
-"* Zeigen Sie, wie Sie `{}` als Platzhalter verwenden müssen, wenn Sie einen Ausdruck (Expression) "
-"drucken möchten,\n"
+"* Zeige, wie Du `{}` als Platzhalter verwenden musst, wenn Du einen Ausdruck (Expression) drucken "
+"möchtest,\n"
 "   der komplexer als nur eine einzelne Variable ist.\n"
 "\n"
-"* Zeigen Sie den Schülern die Standardbibliothek (standard library) und zeigen Sie ihnen, wie sie "
-"nach `std::fmt` suchen\n"
+"* Zeige den Schülern die Standardbibliothek (standard library) und zeige ihnen, wie sie nach `std::"
```
Comment (ronaldfw) [inline po/de.po:2308]:
```suggestion
"* Zeige den Kursteilnehmern die Standardbibliothek (standard library) und zeige ihnen, wie sie nach `std::"
```

File: po/de.po [hunk 12]
```diff
@@ -3199,11 +3264,10 @@ msgstr ""
 "  * Akzeptiert immer eine feste Anzahl von Parametern.\n"
 "  * Akzeptiert immer einen einzelnen Satz von Parametertypen.\n"
 "* Standardwerte werden nicht unterstützt:\n"
-"  * Alle Aufrufseiten haben die gleiche Anzahl von Argumenten.\n"
+"  * Alle Aufrufseiten (call sites) haben die gleiche Anzahl von Argumenten.\n"
```
Comment (ronaldfw) [inline po/de.po:3267]:
```suggestion
"  * Alle Aufrufstellen (call sites) haben die gleiche Anzahl von Argumenten.\n"
```

**Review by ronaldfw — APPROVED** [review]
LGTM, just some minor additional nits I found

File: po/de.po (hunk 12 above)
Comment (mgeisler) [reply po/de.po:3267]: Just a tip: you can use "Add suggestion to batch" on your own suggestions and commit them directly. I often do this for typo fixes where I'm pretty sure that the PR author won't be upset with my change :smile:  After committing my own suggestion like this, I then go ahead and merge the PR. That saves a few round trips for everybody.

**Pull request merged**